pkg/volume/csi/csi_mounter.go

```diff
@@ -199,26 +209,29 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 			nodeName := string(c.plugin.host.GetNodeName())
 			c.publishContext, err = c.plugin.getPublishContext(c.k8s, volumeHandle, string(driverName), nodeName)
 			if err != nil {
-				return err
+				// we could have a transient error associated with fetching publish context
+				opExitStatus = volumetypes.OperationStateNoChange
+				return opExitStatus, err
 			}
 			publishContext = c.publishContext
 		}
 
 	default:
-		return fmt.Errorf("volume source not found in volume.Spec")
+		return opExitStatus, fmt.Errorf("volume source not found in volume.Spec")
 	}
 
 	// create target_dir before call to NodePublish
 	if err := os.MkdirAll(dir, 0750); err != nil {
-		return errors.New(log("mounter.SetUpAt failed to create dir %#v:  %v", dir, err))
+		return opExitStatus, errors.New(log("mounter.SetUpAt failed to create dir %#v:  %v", dir, err))
 	}
 	klog.V(4).Info(log("created target path successfully [%s]", dir))
 
 	nodePublishSecrets = map[string]string{}
 	if secretRef != nil {
 		nodePublishSecrets, err = getCredentialsFromSecret(c.k8s, secretRef)
 		if err != nil {
-			return fmt.Errorf("fetching NodePublishSecretRef %s/%s failed: %v",
+			opExitStatus = volumetypes.OperationStateNoChange
```

Comment:
target_dir will not be cleaned up here?

Reply:
As discussed previously with @jsafrane  - cleaning up of pod's local volume directory is broken regardless of what we do here. I was thinking of fixing this separately in - https://github.com/kubernetes/kubernetes/issues/84878